lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -2349,28 +2436,132 @@ void HdVP2Material::_UpdateShaderInstance(const HdMaterialNetwork& mat)
 
 /*! \brief  Acquires a texture for the given image path.
  */
-const HdVP2TextureInfo&
-HdVP2Material::_AcquireTexture(const std::string& path, const HdMaterialNode& node)
+const HdVP2TextureInfo& HdVP2Material::_AcquireTexture(
+    HdSceneDelegate*      sceneDelegate,
+    const std::string&    path,
+    const HdMaterialNode& node)
 {
     const auto it = _textureMap.find(path);
     if (it != _textureMap.end()) {
         return it->second;
     }
 
-    bool                 isSRGB = false;
-    MFloatArray          uvScaleOffset;
-    MHWRender::MTexture* texture = _LoadTexture(path, isSRGB, uvScaleOffset, node);
+    if (_IsDisabledAsyncTextureLoading()) {
+        bool        isSRGB = false;
+        MFloatArray uvScaleOffset;
+
+        VtValue value;
+        auto    fallbackIt = node.parameters.find(_tokens->fallback);
+        if (fallbackIt != node.parameters.end()) {
+            value = fallbackIt->second;
+        }
+
+        MHWRender::MTexture* texture = _LoadTexture(path, isSRGB, uvScaleOffset, value);
+
+        HdVP2TextureInfo& info = _textureMap[path];
+        info._texture.reset(texture);
+        info._isColorSpaceSRGB = isSRGB;
+        if (uvScaleOffset.length() > 0) {
+            TF_VERIFY(uvScaleOffset.length() == 4);
+            info._stScale.Set(
+                uvScaleOffset[0], uvScaleOffset[1]); // The first 2 elements are the scale
+            info._stOffset.Set(
+                uvScaleOffset[2], uvScaleOffset[3]); // The next two elements are the offset
+        }
+
+        return info;
+    }
+
+    VtValue fallbackValue;
+    auto    fallbackIt = node.parameters.find(_tokens->fallback);
+    if (fallbackIt != node.parameters.end()) {
+        fallbackValue = fallbackIt->second;
+    }
+
+    {
+        std::lock_guard<std::mutex> lock(_textureLoadingMutex);
+        _textureLoadingTasks.emplace(
+            path, new TextureLoadingTask(this, sceneDelegate, path, fallbackValue));
+    }
+
+    static HdVP2TextureInfo dummyInfo;
+
+    if (!dummyInfo._texture) {
+        // Create a default texture info with fallback color
+        MHWRender::MRenderer* const       renderer = MHWRender::MRenderer::theRenderer();
+        MHWRender::MTextureManager* const textureMgr
+            = renderer ? renderer->getTextureManager() : nullptr;
+        if (textureMgr) {
+            GfVec4f fallbackColor { 0.8, 0.8, 0.8, 0 };
+            if (fallbackIt != node.parameters.end() && fallbackIt->second.IsHolding<GfVec4f>()) {
+                fallbackColor = fallbackIt->second.UncheckedGet<GfVec4f>();
+            }
+            dummyInfo._texture.reset(
+                _GenerateFallbackTexture(textureMgr, "dummy", VtValue(fallbackColor)));
+        }
+    }
+
+    return dummyInfo;
+}
+
+void HdVP2Material::LoadTextures()
```

Comment:
This function should be renamed something like enqueueLoadTextures or something so that the reader knows they will not be loaded immediately.

Reply:
Done.